### PR TITLE
SelectableRegion does not merge child semantics nodes

### LINKED ISSUE #104557: `SelectionArea` breaks for TalkBack
## Description 
Wrapping a `SelectionArea` widget around multiple Text widgets breaks the selection for TalkBalk. TalkBalk is still able to read the text but I'm unable to select a `Text` widget inside the tree.

### With `SelectionArea`
https://user-images.githubusercontent.com/24459435/170145265-6f4b3b4b-2911-4912-818e-056f2238f598.MOV

### Without `SelectionArea` (how it should be)
https://user-images.githubusercontent.com/24459435/170145239-a84091bb-135d-466a-bf7f-359861b07287.MOV

I used a Google Pixel 4a with Android.

<details>
<summary>My app</summary>

```dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// This example shows how to make a screen selectable..

import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      // You need to add the "SelectionArea" to make the text selectable.
      home: SelectionArea(
        child: Scaffold(
          body: Padding(
            padding: const EdgeInsets.all(16),
            child: Center(
              child: Column(
                mainAxisSize: MainAxisSize.min,
                children: <Widget>[
                  Text(
                    'Fast',
                    style: Theme.of(context).textTheme.titleLarge,
                  ),
                  const Text(
                    'Flutter code compiles to ARM or Intel machine code as well as JavaScript, for fast performance on any device.',
                  ),
                  ElevatedButton(
                    onPressed: () {},
                    child: const Text('Try it in DartPad'),
                  )
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

cc: @chunhtai I think this should get a higher priority because TalkBalk is an import accessibility feature.

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/104557

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by nilsreichardt — COMMENTED** [review]
I can confirm the fix 👍 

https://user-images.githubusercontent.com/24459435/170477999-08559b00-4b16-4b17-bfbf-08421dd173fa.MOV

File: packages/flutter/test/widgets/selectable_region_test.dart [hunk 1]
```diff
@@ -94,6 +95,75 @@ void main() {
...
+                        TestSemantics(
+                          label: 'Line one',
+                          textDirection: TextDirection.ltr,
+                        ),
+                        TestSemantics(
+                          label: 'Line two',
+                          textDirection: TextDirection.ltr,
+                        ),
+                        TestSemantics(
+                          flags: <SemanticsFlag>[SemanticsFlag.isButton,
```
Comment (goderbauer) [inline packages/flutter/test/widgets/selectable_region_test.dart:144]: nit: maybe put these on separate lines to make it a little easier to read.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/test/widgets/selectable_region_test.dart [hunk 2]
```diff
@@ -94,6 +95,75 @@ void main() {
...
+          ],
+        ),
+        ignoreRect: true,
+        ignoreTransform: true,
+        ignoreId: true,
+      ),
+    );
+
+      semantics.dispose();
+    });
```
Comment (goderbauer) [inline packages/flutter/test/widgets/selectable_region_test.dart:165]: nit: somehow the indentation looks off here.

**Pull request merged**